core-web/libs/sdk/editor/src/lib/sdk-editor.ts

```diff
@@ -0,0 +1,420 @@
+import { CUSTOMER_ACTIONS, postMessageToEditor } from './models/client.model'
+import { NOTIFY_CUSTOMER } from './models/editor.model';
+import { findContentletElement, getClosestContainerData, getPageElementBound } from './utils/editor.utils';
+
+interface DotCMSPageEditorConfig {
+    onReload: () => void;
+}
+class DotCMSPageEditor {
+    private config: DotCMSPageEditorConfig;
+    isInsideEditor!: boolean;
+
+    constructor(config?: DotCMSPageEditorConfig) {
+        this.config = config ?? { onReload: defaultReloadFn };
+    }
+
+    init() {
+        console.log('SdkDotPageEditor Headless init!!');
+        this.isInsideEditor = this.checkIfInsideEditor();
+        if (this.isInsideEditor) {
+            this.listenEditorMessages();
+            this.listenHoveredContentlet();
+            this.scrollHandler();
+            this.listenContentChange(); // We can use const observer = listenContentChange() to disconnect later
```

Comment:
We need to find a scalable way to manage events and observers to make a `destroy` method.

Maybe something like:

```javascript
[{
    event: "scroll",
    element: window, 
    callback: myScrollHandler, // We can also add another property as beforeCleanUp, that serve as an internal memory cleaner
    type: "DOM_EVENT"
},
{
    type: "OBSERVER",
    reference: myObserver
    disconnect: myObserver.disconnect
}]
```

And all the time we add a listener we just push to that array or something like that, so the destroy method is written once and only modified in case we need to add more types or handle more logic

Reply:
We can also do something similar for the `init `method, like add callbacks to an array and then iterate through it calling every method needed in the `init()`, instead of manually adding it to the `init()`. Can be useful if we need to add more logic before calling something else as `beforeInit` method

Reply:
The philosophy behind this is to make the code more close to modifications and open to extension, this way when we need to add more functionality we don't need to modify the `init` or the `destroy` in order to make our code work.

Reply:
Also this can establish an standard in how we write the events and how they get initialized and cleaned

Reply:
Yep we must to unsubscribe the listeners, i just add the logic [here](https://github.com/dotCMS/core/pull/27717/files#diff-4bb8ee6eec27c199d731de8321816d115804c2129f6d08c10e4e8b11b8f27868)

But i dont totally like to add the listeners in array iterations
I feel the manually call "listenXXXX()" is more easy to understand

thoughts? cc: @fmontes @rjvelazco